cluster-autoscaler/proposals/cp_agnostic_ca.md

```diff
@@ -0,0 +1,158 @@
+# Decoupling ClusterAutoScalerCore (separate library) from CloudProvider implementations
+##### Author: bhargav-naik
+
+### Introduction
+Current ClusterAutoscaler is released as a bundle which includes core and all the CloudProviders included.
+When someone intends to write implementation of the CloudProvider they need to fork the intended version ClusterAutoscaler and provide their implementation.
+In this proposal, a different design approach is presented which will decouple the ClusterAutoscaler core from CloudProviders and make ClusterAutoScalerCore available as a separate release. 
+
+### Issues
+- In the current design the ClusterAutoscaler core has dependency on the CloudProvider implementations.
+- For proprietary CloudProvider implementations it results in an development cycle to fork new version from public repo and port the CloudProvider implementation to the latest fork.
```

Comment:
This is not true as mentioned in the gRPC proposal, you could use the new cluster-api provider

Reply:
- Agreed, using cluster-api is definitely an option external_cp must consider.
- But CA might not want to enforce such user to go via CAPI path. CA should still have a way to enable external_cp without having them use CAPI.